sql-plugin/src/main/scala/org/apache/spark/sql/catalyst/json/rapids/GpuJsonScan.scala

```diff
@@ -315,15 +315,13 @@ class JsonPartitionReader(
       getFileFormatShortName, partFile)
     withResource(jsonTbl) { tbl =>
       val columns = new ListBuffer[ColumnVector]()
-      closeOnExcept(columns) { _ =>
-        for (name <- readDataSchema.fieldNames) {
-          val i = cudfSchema.getColumnNames.indexOf(name)
-          if (i == -1) {
-            throw new IllegalStateException(
-              s"read schema contains field named '$name' that is not in the data schema")
-          }
-          columns += tbl.getColumn(i)
+      for (name <- readDataSchema.fieldNames) {
```

Comment:
Nit: Rather than an explicit listbuffer and iterating on fieldNames (which itself is a .map on the schema), we could just .map on the schema directly like this (which also removes the wasteful building of the column names array each time):
```
val cudfColumnNames = cudfSchema.getColumnNames
val columns = readDataSchema.map { f =>
  val i = cudfSchema.getColumnNames.indexOf(f.name)
  if (i == -1) {
    throw ..
  }
  tbl.getColumn(i)
}
```

Reply:
will push shortly

Reply:
Handled here: https://github.com/NVIDIA/spark-rapids/pull/8413/commits/13d08ca8a616ee5fe0f7229d33bd2c729e9ed20f